airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -119,18 +124,24 @@ const getOptimalSyncMode = (
 const calculateInitialCatalog = (
   schema: SyncSchema,
   supportedDestinationSyncModes: DestinationSyncMode[],
-  isNotCreateMode?: boolean
-): SyncSchema => ({
-  streams: schema.streams.map<SyncSchemaStream>((apiNode, id) => {
-    const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
-    const nodeStream = verifySourceDefinedProperties(verifySupportedSyncModes(nodeWithId), isNotCreateMode || false);
-
-    if (isNotCreateMode) {
-      return nodeStream;
-    }
-
-    return getOptimalSyncMode(verifyConfigCursorField(nodeStream), supportedDestinationSyncModes);
-  }),
-});
+  isNotCreateMode?: boolean,
```

Comment:
nitpick: maybe a matter of preference, but I find negative names for booleans harder to read - in this case I would prefer `!isCreateMode` to `isNotCreateMode`